play/tests/test_jamesk.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+
+from play.jamesk import knockknock
+
+def test_knockknock():
+    import sys
+    import io
+    import re
+
+    # redirect stdout to a string stream
```

Comment:
Best done with the stdlib `contextlib.redirect_stdout` (https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stdout).

Reply:
This is great!  Definitely simplified redirecting stdio.  added in 37b20472ed441bf813243e3daadb605296cf3d64.